pkg/ruler/api.go

```diff
@@ -0,0 +1,264 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/weaveworks/common/user"
+)
+
+// RegisterRoutes registers the configs API HTTP routes with the provided Router.
+func (r *Ruler) RegisterRoutes(router *mux.Router) {
+	if r.store == nil {
+		level.Info(util.Logger).Log("msg", "ruler configured with store that does not support api")
+		return
+	}
+	for _, route := range []struct {
+		name, method, path string
+		handler            http.HandlerFunc
+	}{
+		{"get_rules", "GET", "/api/prom/api/v1/rules", r.rules},
+		{"get_alerts", "GET", "/api/prom/api/v1/rules", r.alerts},
+	} {
+		level.Debug(util.Logger).Log("msg", "ruler: registering route", "name", route.name, "method", route.method, "path", route.path)
+		router.Handle(route.path, route.handler).Methods(route.method).Name(route.name)
+	}
+}
+
+// In order to reimplent the prometheus rules API, a large amount of code was copied over
+// This is required because the prometheus api implementation does not pass a context to
```

Comment:
I'm a bit dubious. I was expecting the following code to be a close replica of https://github.com/cortexproject/cortex/blob/master/vendor/github.com/prometheus/prometheus/web/api/v1/api.go but it's actually quite different.

Is it because it has been copied long time ago? If so, would it make sense to redo the copy-paste? Just asking to better undestand: don't want you to do extra work.

Reply:
The main things that were copied over was the structs. Most of the logic for handling the requests is simplified versions of what was in Prometheus.